app/scripts/models/auth_brokers/fx-desktop.js

```diff
@@ -65,6 +71,36 @@ define([
           self, options);
     },
 
+    /**
+      * Is used to ask the client about capabilities before rendering the views
+      *
+      * @returns {Promise}
+      */
+    beforeRender: function () {
+      var self = this;
+      var defer = p.defer();
+      var result = null;
+
+      self.request(self._commands.CAPABILITIES)
```

Comment:
The naming of `beforeRender` here suggests that this will be called every time a view is rendered, which could be a lot.  ISTM that we should do this only once, during page load and initialization, as a kind of "handshake" with the device.